### PR TITLE
Artist Model 

### PR DESCRIPTION
**Pull request by TOMalley104**
Creates Artist class that encapsulates SPTPartialArtist and SPTArtist

- refactors Spotify manager and all VCs to use Artist instead of SPTPartial/FullArtist
- removes Spotify manager involvement in VCs past search (once the VCs have Artist objects, they rely solely on those)
- Artists update themselves (full info, top tracks) and post a notification when complete. VCs listen for notifications and update views as necessary. 
- Add `toNSError` and `init?(error:NSError)` to SpotifyError class to mitigate sending errors through notifications. ***

*** This part is dicey and will incite the most discussion. To create `.toNSError()` I basically ripped out the code from `presentErrorAlert` that creates a message for each SpotiftyError case. Then with the intent of not repeating myself, in `presentErrorAlert` I convert the SpotifyError to an NSError and use its description as the alert message.

### CODE REVIEW

File: Podfile.lock [hunk 1]
```diff
@@ -20,17 +20,19 @@ PODS:
     - FontAwesomeKit/Core
   - FontAwesomeKit/Zocial (2.2.1):
     - FontAwesomeKit/Core
-  - Intrepid (0.3.1):
-    - Intrepid/Core (= 0.3.1)
-  - Intrepid/Core (0.3.1):
+  - Intrepid (0.3.2):
```
Comment (brightredchilli) [inline Podfile.lock]: do me a favor, and do a pod update Intrepid one more time here. there was some weird things happening with the pod yesterday, you should be on 0.3.3 instead.

File: spotifyMini/SpotifyError.swift [hunk 2]
```diff
@@ -8,22 +8,49 @@
...
+let SpotifyMiniErrorLogInFailedCode = 123
+let SpotifyMiniErrorInvalidSessionCode = 666
+let SpotifyMiniErrorRequestFailed = 420
+
 enum SpotifyError : ErrorType {
     case LogInFailed
     case InvalidSession
     case RequestFailed
+
+    func toNSError() -> NSError {
```
Comment (brightredchilli) [inline spotifyMini/SpotifyError.swift]: the more swift way of doing this is to make a convenience initializer on NSError instead.

File: spotifyMini/SpotifyError.swift [hunk 3]
```diff
@@ -8,22 +8,49 @@
 
 import Foundation
 
+let SpotifyMiniErrorDomain = "com.tomalley.spotifyMini"
+let SpotifyMiniErrorLogInFailedCode = 123
```
Comment (brightredchilli) [inline spotifyMini/SpotifyError.swift]: where do these numbers come from, are they just arbitary?

**Comment by brightredchilli** [comment]
:ship:
Should definitely squash some of these commits down. for example(small refactor to artist is probably not a necessary detail for the future)

File: spotifyMini/SpotifyError.swift (hunk 3 above)
Comment (TOMalley104) [inline spotifyMini/SpotifyError.swift]: yep, figured as long as they were constantized it wouldn't be a problem. is it?

**Pull request merged**